amethyst_audio/src/components/audio_emitter.rs

```diff
@@ -53,3 +54,79 @@ impl AudioEmitter {
 impl Component for AudioEmitter {
     type Storage = BTreeStorage<Self>;
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::{AudioEmitter, Source};
+    use amethyst_utils::app_root_dir::application_root_dir;
+    use std::{fs::File, io::Read, vec::Vec};
+
+    // test_play tests the AudioEmitter's play function
+    fn test_play(file_name: &str, should_pass: bool) {
+        // Get the full file path
+        let app_root = application_root_dir().unwrap();
+        let audio_path = app_root.join(file_name);
+
+        // Convert the file contents into a byte vec
+        let mut f = File::open(audio_path).unwrap();
+        let mut buffer = Vec::new();
+        f.read_to_end(&mut buffer).unwrap();
+
+        // Create a Source and AudioEmitter from those bytes
+        let src = Source { bytes: buffer };
+        let mut emitter = AudioEmitter::default();
+
+        // Call play
+        match emitter.play(&src) {
+            Ok(_pass) => assert!(should_pass),
+            Err(_fail) => assert!(!should_pass),
```

Comment:
for the it-failed-but-should-not-fail case, since the error type contains information, perhaps do something like this:

```rust
e @ Err(_fail) => assert!(!should_pass, "Expected `play` result to be `Ok(..)`, but was {:?}", e),
```

(syntax might be wrong, but there's the idea)

Reply:
I added more useful commentary.

Reply:
ah, could you either use `e`, or remove the underscore from `_fail` (underscore means it's unused).

Also, could you use the mp3 and wav from https://github.com/mathiasbynens/small ? (gotta rebase the bigger files out of history)
Thanks!

Reply:
Removed all instances of the fail underscores. ty for catching it.